### PR TITLE
25685 and 25689, justification box validations

### PR DESCRIPTION
**Pull request by JoeCatling**

### CODE REVIEW

File: app/models/cas_application_fields.rb [hunk 1]
```diff
@@ -17,4 +21,26 @@ def declaration_choices
       end
     end.compact.to_h
   end
+
+  def reason_justification_must_be_populated_if_level_1_selected
+    return if reason_justification.present?
+    return unless project.project_datasets.any? do |pd|
```
Comment (kenny-lee-1) [inline app/models/cas_application_fields.rb:27]: I think you've already got a direct association on project i.e 
`project.project_dataset_levels.where(access_level_id: Lookups::AccessLevel.find_by(value: 1)).count.positive?`

File: app/models/cas_application_fields.rb [hunk 2]
```diff
@@ -17,4 +21,26 @@ def declaration_choices
...
+                      pdl.access_level_id == 1 && pdl.selected == true
+                    end
+                  end
+
+    errors.add :reason_justification, :must_be_present
+  end
+
+  def extra_datasets_rationale_must_be_populated_if_extra_datasets
+    return if extra_datasets_rationale.present?
+    return unless project.project_datasets.any? do |pd|
```
Comment (kenny-lee-1) [inline app/models/cas_application_fields.rb:38]: as above, can use the project association

**Review by kenny-lee-1 — CHANGES_REQUESTED** [review]
just a couple of association querying changes!

File: app/models/cas_application_fields.rb (hunk 1 above)
Comment (JoeCatling) [reply app/models/cas_application_fields.rb:27]: project_dataset_levels association doesn’t work with the project helper that we are using to build all possible levels in memory in the form. Therefore, the levels and associated project_datasets that we want to evaluate here aren't persisted at the point of validation.

File: app/models/cas_application_fields.rb (hunk 2 above)
Comment (JoeCatling) [reply app/models/cas_application_fields.rb:38]: project_dataset_levels association doesn’t work with the project helper that we are using to build all possible levels in memory in the form. Therefore, the levels and associated project_datasets that we want to evaluate here aren't persisted at the point of validation.

**Review by kenny-lee-1 — APPROVED** [review]

**Pull request merged**